server/routes/api.js

```diff
@@ -236,17 +239,120 @@ router.post('/event', passport.authenticate('jwt', {session: false}), async (req
 
   // Add the creator as an attendee in the back-end
   const member_event = new Member_Event({
-    date: new Date(),
+    date: [new Date()],
     paid: false,
     member: req.body.creatorId,
-    event: event._id
+    event: event._id,
+    tickets: 0
   });
   member_event.save()
     .catch(err => {
       console.log(err);
   });
 });
 
+router.post('/ticket',passport.authenticate('jwt', {session: false}), async (req, res)=>{
+    try{
+        const { userId, eventId } = req.body;
+
+        const event = await Event.findById(eventId);
+        const user = await Member.findById(userId);
+
+        if(!event || !user){
+            return res.status(404).json({ error: 'User or Event not found' });
+    
+        }
+        const event_member = await Member_Event.findOne({
+            member: user._id,
+            event: event._id
+        });
+        if(event.tickets === event.ticketsSold){
+            return res.status(409).json({ error: 'There is no tickets left' }); 
+        }
+        if(event_member && event_member.paid){
+          return res.status(409).json({ error: 'User already have a ticket' });
+        }
+
+        event.ticketsSold++;
+
+
+        if(event_member){
+            if(!event_member.paid){
```

Comment:
You could also add here that the date of the even_member entry is updated to the current date :D so just date: new Date()